### ISSUE TITLE
Support automated ropensci list & blacklisted packages

### ISSUE THREAD

**Issue by cboettig** [cboettig/drat#3]
currently `ropensci.txt` is modified by hand, not the latest list from ropensci.R / ropkgs list.  Need to find out why the packages commented out from that list don't build and fix.

**Comment by cboettig** [comment]
@sckott has mostly addressed this in ropkgs, but there's a few that don't seem to build; not sure if that's a drat issue, a drat.builder issue or an external dependencies issue:

currently looks like these are:

```
ropensci/rdopa/
ropensci/RedisAPI/
ropensci/rrlite/
```

(I think two of those are only on the list because they depend on the other one, but I forget which is which).  Guess I gotta test these out still.

**Comment by sckott** [comment]
rrlite depends on RedisAPI i believe

**Comment by sckott** [comment]
what's the error with `rdopa`? it does depend on `rgeos` https://github.com/ropensci/rdopa/blob/master/DESCRIPTION#L19, which could cause a problem, as it's an external C dep

**Comment by cboettig** [comment]
I'll take a look.  in principle I should have all those external libraries
already installed on rocker/ropensci:dev image, but maybe not.  This should
help me catch what I've missed there.

On Wed, Jun 24, 2015 at 11:51 AM Scott Chamberlain <notifications@github.com>
wrote:

> what's the error with rdopa? it does depend on rgeos
> https://github.com/ropensci/rdopa/blob/master/DESCRIPTION#L19, which
> could cause a problem, as it's an external C dep
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cboettig/drat/issues/3#issuecomment-114978922>.
>
-- 

http://carlboettiger.info

**Comment by cboettig** [comment]
@sckott appears to be the vignette on rdopa that is failing.  https://circleci.com/gh/cboettig/drat/123

**Comment by cboettig** [comment]
RedisAPI also seems to fail on vignette step: https://circleci.com/gh/cboettig/drat/124

**Comment by sckott** [comment]
@cboettig I'll ask the rdopa maintainer to fix that

@richfitz can you fix the RedisAPI vignette?

**Comment by richfitz** [comment]
It just needs redis server running.  I might have to change the vignette to not build on cran.  Failing that, we can add the `{"vignettes": false}` flag to the RedisAPI line...

**Comment by sckott** [comment]
hmm, but can't you add redis to travis, http://docs.travis-ci.com/user/database-setup/#Redis

**Comment by richfitz** [comment]
Sorry? That _is_ how you add Redis to travis: https://github.com/ropensci/RedisAPI/blob/master/.travis.yml#L12-L14

Not sure how on circle though.

**Comment by cboettig** [comment]
it would be the same on circle, except that the build is happening in a container so it doesn't care what services you've turned on outside the container. I'd need to add and start a redis-server running on the rocker/ropensci image.  

I'm surprised you can get away with assuming a redis server is running locally -- does CRAN really let you build a vignette that way?  Usually it's sufficient to check the suggests list and system dependencies in order to build the vignette.  Recall the ropensci.txt file read by drat.builder is generated automatically and I don't see any way of programmatically discovering that this package has some feature for which we should trigger the no-build-vignettes option.

**Comment by richfitz** [comment]
It's not on CRAN, and no it looks like it won't fly there.

But it does seem like a reasonable thing to me: if you're going to build the vignette about Redis, you're going to need Redis running...

**Comment by cboettig** [comment]
@richfitz I'm not saying it isn't reasonable, I'm just saying that computers don't do what's reasonable, they just do what they are told.  

Why not do what we used to do in the days before knitr was a valid vignette engine and just build the vignette from the true sources in inst.  You can generate an output .md file (like in the case with README.Rmd) and then rename it with .Rmd so that the vignette builder thinks it is a vignette source document, even though it has no code blocks.

Sure that's a hack, but your 'reasonable' user can always figure out that they have to start a redis server and then use the vignette sources or copy paste from the vignette or whatever they like to do.  Meanwhile the dumb computer trying to build the package doesn't have to guess that it needs a Redis server running.

**Comment by richfitz** [comment]
I agree, I just haven't got around to it, partly because I don't know how we used to things!  Could you do me a favour and make a PR showing how this is done?  Once that's done I can get it on CRAN.

**Comment by cboettig** [comment]
I can give it a go, assuming I can figure out how to get the vignette to
build locally ;-)


On Wed, Jun 24, 2015 at 3:24 PM Rich FitzJohn <notifications@github.com>
wrote:

> I agree, I just haven't got around to it, partly because I don't know how
> we used to things! Could you do me a favour and make a PR showing how this
> is done? Once that's done I can get it on CRAN.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cboettig/drat/issues/3#issuecomment-115029339>.
>
-- 

http://carlboettiger.info

**Comment by cboettig** [comment]
https://github.com/ropensci/RedisAPI/pull/6

p.s. since when did GitHub stop folding quoted text from email threads into a `...` on GitHub issues?
